### DISCUSSION TITLE
How do you view, re-create or otherwise operate the database?

### DISCUSSION THREAD

**Discussion by kimmobrunfeldt** [vlcn-io/cr-sqlite#429] [Q&A]
I'm having a hard time figuring out the best way to access the SQLite database for debugging purposes. For example I accidentally wrote incorrect data and would need to clear the database completely to start over. How would you do that?

Tried to also look for any files in the OPFS using [OPFS Explorer](https://tomayac.github.io/opfs-explorer/) but found none:

![Screenshot 2024-04-13 at 9 20 23](https://github.com/vlcn-io/cr-sqlite/assets/1232405/64799f56-e12b-49a8-838b-abac76fd0220)

I'm using `@vlcn.io/react` so the application gets the database context handler via hooks. Exposing that to e.g. dev tools didn't work by doing:

```tsx
  const ctx = useDB(DATABASE_NAME)
  ;(globalThis as any).ctx = ctx
```
not sure why.

How do you usually do this kind of debugging? Have you been able to connect any SQLite GUIs to the database? For example https://sqlitebrowser.org/

**Comment by kimmobrunfeldt** [comment]
I eventually figured out that the data is written to IndexedDB

**Reply by tantaman** [reply]
It is also possible to use `OPFS` although it isn't documented on how to set that up. You can check out the `wa-sqlite` repository and their examples of using OPFS.
